docs/rfcs/0041-progressive-rollout.md

```diff
@@ -0,0 +1,291 @@
+# Summary
+
+# UI
+
+We'll introduce two kinds of UI, "Manual Setting" and "Template Setting".
+
+## Manual Setting
+
+Users can determine any percentage and schedule in this UI.
+
+![manual-setting-proto-ui](./images/0041-image1.png)
+
+## Template Setting
+
+Manual Setting has a lot of flexibility, but it’s hard when configuring a periodic increase in the percentage of a variation.
+In this case, Template Setting is useful.
+
+![template-setting-proto-ui](./images/0041-image2.png)
+
+## Progress bar of rollout
+
+![progress-bar-of-rollout](./images/0041-image4.png)
+
+# Important Notice
+
+* Users can use Progressive Rollout when the number of variations is equal to 2.
+* Users can't use same scheduled time in single auto ops rules. For example, users can not set true for 50% at 2023-01-01 00:06:00 and 80% at the same time.
+* The interval of time for each scheduled time must be at least 5 minutes.
+* Users can stop Progressive Rollout temporary.
+* Users can use both Progressive Rollout and Feature Flag Trigger at the same time.
+* The operation of Progressive Rollout is done regardless of whether the feature flag is enabled or disabled.
+
+# Processing flow
+
+The following image is a processing flow in this feature.
+
+![processing-flow](./images/0041-image3.png)
+
+For instance, Web Client sends clause as follows:
+
+```go
+&autoopsproto.ProgressiveRolloutClause{
+	// The another varition id is vid-2
+	VariationId: "vid-1",
+	Schedules: []*autoopsproto.ProgressiveRolloutClause_Schedule{
+		{
+			// '2023-01-01 00:03:00'
+			Time: 1672498980,
+			Weight: 20000,
+		},
+		{
+			// '2023-01-01 00:06:00'
+			Time: 1672499160,
+			Weight: 40000,
+		},
+		{
+			// '2023-01-01 00:09:00'
+			Time: 1672499340,
+			Weight: 60000,
+		},
+	},
+}
+```
+
+1. Web Client registers the above clauses by calling `CreateProgressiveRollout` rules
+2. Batch service calls `ListProgressiveRollout`, and check if the current time is a scheduled time. In this case, it checks whether the current time is 2023-01-01 00:03:00.
+3. If the current time is a scheduled time and the rule is enabled, Batch service call `ExecuteProgressiveRollout`.
+4. AutoOps service calls `UpdateFeatureTargeting` to update feature rules. In this case, update the weight of vid-1 to 20000 and the weight of vid-2 to 80000.
+
+# Changes
+
+## Table
+
+We'll create `ops_progressive_rollout` table as follows. `ProgressiveRolloutManualScheduleClause` and `ProgressiveRolloutAutomaticScheduleClause` are converted into Any type and stored into `clause` column.
+
+```sql
+CREATE TABLE IF NOT EXISTS `ops_progressive_rollout` (
```

Comment:
Please add the type field to the table definition as described here.
https://github.com/bucketeer-io/bucketeer/pull/282/files#r1166241238